app/controllers/concerns/provisioners.rb

```diff
@@ -0,0 +1,196 @@
+# frozen_string_literal: true
+
+# module to manage all the provisioners logic in the deploy controller
+module Provisioners
+  include I18n
+
+  # Interanl syntax used to identify the provisioner, E.g. hana_provision_1
+  PROVISIONER_NAME_PATTERN = /(.*)_(\d+)/.freeze
+  # EXCLUDED_PATTERN filters the terraform entries that won't go in the general
+  # infrastructure creation bar, as they are the provisioning resources
+  EXCLUDED_PATTERN = /.*\.(.*_provision).*\.provision\[(\d+)\]?/.freeze
+  PROVISIONING_STATES = {
+    not_started:    I18n.t('provisioning_states.not_started'),
+    initializing:   I18n.t('provisioning_states.initializing'),
+    configuring_os: I18n.t('provisioning_states.configuring_os'),
+    provisioning:   I18n.t('provisioning_states.provisioning'),
+    failed:         I18n.t('provisioning_states.failed'),
+    finished:       I18n.t('provisioning_states.finished')
+  }.freeze
+
+  PROVISIONING_PATTERNS = {
+    planned_states_count: Rails.configuration.x.provisioning_patterns['planned_states_count'],
```

Comment:
what happens here in case `terraform` output is configured in some asiatic/japan language or something similar?

Reply:
The whole project is under our control. This piece of code will be used only for our deployment. 
We control de images and the installed software.
We are not going to have external agents here changing this configuration.